### ISSUE TITLE
toolmaker uses wrong animation for delivering tools

### ISSUE THREAD

**Issue by stefson** [Return-To-The-Roots/s25client#1056]
It is roughly at gameframe 370400, the toolmaker is right next to my own headquarter: 

![2019-03-23-140726_1680x1050_scrot](https://user-images.githubusercontent.com/14232936/54866542-630fc100-4d75-11e9-8469-9b3097e30d17.png)

replay: 
[2019-03-23_13-50-54.rpl.zip](https://github.com/Return-To-The-Roots/s25client/files/2999359/2019-03-23_13-50-54.rpl.zip)

actuall consequences: none, the game doesn't crash and the boat is transformed into a bow upon dropping the boat at the flag :)

**Comment by Flamefire** [comment]
I can't find the graphic for the guy carrying a bow in JOBS.BOB. Is it possible that it is missing? What did the original do? @Spikeone

**Comment by Flow86** [comment]
didn't we have an extra drop-in graphic for that? I think the original did also carry a boat, or did he carry a sword instead in the original?

**Comment by Flamefire** [comment]
Doesn't look like we have that.

I changed it locally to a carrier with a bow. Looks ok: 
![Bildschirmfoto vom 2019-05-14 12-54-38](https://user-images.githubusercontent.com/309017/57753059-5bbcc200-76eb-11e9-8a36-357ba7cfa9d3.png)

Compare to regular worker with thongs:
![Bildschirmfoto vom 2019-05-14 12-44-46](https://user-images.githubusercontent.com/309017/57753080-64ad9380-76eb-11e9-96fd-ec00ec8aac38.png)

What do you say?

**Comment by Flow86** [comment]
hmm, I don't like it, its kind of a hack. Perhaps @Spikeone could create those 4 or 6 missing pics and we could fix it the "right way"?

**Comment by Flamefire** [comment]
It would be ~16 pics (2 directions a 8 animation steps) but in BOB format which is hard to handle (not sure we can pack this)

**Comment by Flow86** [comment]
we do not need to pack them. the loader was able to handle that. you only have to name them properly (index ...)

**Comment by Spikeone** [comment]
@Flamefire @Flow86 will do - would also like having "correct" graphics which should be not that hard in this case.

**Comment by Flamefire** [comment]
> we do not need to pack them. the loader was able to handle that. you only have to name them properly (index ...)

Only if we replace existing ones for BOB files. The format there is tricky: You give it an ID and it assembles an image from a body and a "head" (including the item)

As far as I can tell 2263-2273 (11 images) can be used. It has the ID 92 and was probably intended for producing buckets for the well guy.

@Spikeone You'll probably get away by just creating 2 files (front&back view) and duplicate that. I'll push an update which uses these indices so you can test it by putting them into RTTR/LSTS/GAME/JOBS.BOB

**Comment by Flow86** [comment]
we can either override images (i.e the well guy which is not used) or use indeces beyond the end of the file (should work too)

**Comment by Flamefire** [comment]
No. This is what I meant: BOB files are not LST files where you have 1 image per IDX. Instead you have 1 image and a mapping to another image. And this mapping can currently neither be appended nor overwritten

**Comment by Spikeone** [comment]
I took a look at those graphics. There are actually graphics with a bow - JOBS.BOB -> 1977 to 1990. At least those are the tool graphics which are also used by carriers.

So no need to create new graphics :)

**Comment by Flamefire** [comment]
Do you mean:
![1977 player nx16 ny19](https://user-images.githubusercontent.com/309017/60325583-edc11700-9987-11e9-87be-49a4a86c387f.png)

Because the worker is completely different
